onnxscript/function_libs/torch_aten/registration.py

```diff
@@ -0,0 +1,70 @@
+"""Registry for aten functions."""
+
+from __future__ import annotations
+
+from typing import Any, Optional
+
+
+class OverloadedFunction:
+    """Overloaded function."""
+
+    def __init__(self, name: str):
+        self.name = name
+        self.default: Optional[Any] = None
+        self.overloads: list[Any] = []
+
+
+class Registry:
```

Comment:
Code gen should then read the registry to generate individual symbolic functions that dispatches to onnx functions & adds glue nodes. So users don't need to interact with this registry, which tends to be opaque.